# Data in Motion Weekly Data Analysis Challenge
## Week 4: U.S. Baby Names

Link to dataset: https://buff.ly/3EtASMF 

Challenge Questions:
1. See the first 10 entries.
2. Delete the columns 'Unnamed: 0' and 'Id'.
3. Group the dataset by name, assign to a variable called names, and sort the dataset by highest to lowest count.
4. How many different names exist in the dataset?
5. What is the name with most occurrences?
6. What is the standard deviation of count of names?
7. Get a summary of the dataset with the mean, min, max, std and quartiles.

In [1]:
# import libraries

import pandas as pd
import numpy as np

In [2]:
# read in data:
df = pd.read_csv('https://buff.ly/3EtASMF')

In [3]:
# view head of data
df.head()

Unnamed: 0     Id     Name  Year Gender State  Count
0       11349  11350     Emma  2004      F    AK     62
1       11350  11351  Madison  2004      F    AK     48
2       11351  11352   Hannah  2004      F    AK     46
3       11352  11353    Grace  2004      F    AK     44
4       11353  11354    Emily  2004      F    AK     41

In [4]:
# view info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1016395 entries, 0 to 1016394
Data columns (total 7 columns):
 #   Column      Non-Null Count    Dtype 
---  ------      --------------    ----- 
 0   Unnamed: 0  1016395 non-null  int64 
 1   Id          1016395 non-null  int64 
 2   Name        1016395 non-null  object
 3   Year        1016395 non-null  int64 
 4   Gender      1016395 non-null  object
 5   State       1016395 non-null  object
 6   Count       1016395 non-null  int64 
dtypes: int64(4), object(3)
memory usage: 54.3+ MB


### 1. See the first 10 entries.

In [5]:
df.head(10)

Unnamed: 0     Id      Name  Year Gender State  Count
0       11349  11350      Emma  2004      F    AK     62
1       11350  11351   Madison  2004      F    AK     48
2       11351  11352    Hannah  2004      F    AK     46
3       11352  11353     Grace  2004      F    AK     44
4       11353  11354     Emily  2004      F    AK     41
5       11354  11355   Abigail  2004      F    AK     37
6       11355  11356    Olivia  2004      F    AK     33
7       11356  11357  Isabella  2004      F    AK     30
8       11357  11358    Alyssa  2004      F    AK     29
9       11358  11359    Sophia  2004      F    AK     28

### 2. Delete the columns 'Unnamed: 0' and 'Id'.

In [6]:
df.drop(columns=['Unnamed: 0', 'Id'], inplace=True)
df

Name  Year Gender State  Count
0           Emma  2004      F    AK     62
1        Madison  2004      F    AK     48
2         Hannah  2004      F    AK     46
3          Grace  2004      F    AK     44
4          Emily  2004      F    AK     41
...          ...   ...    ...   ...    ...
1016390     Seth  2014      M    WY      5
1016391  Spencer  2014      M    WY      5
1016392     Tyce  2014      M    WY      5
1016393   Victor  2014      M    WY      5
1016394   Waylon  2014      M    WY      5

[1016395 rows x 5 columns]

### 3. Group the dataset by name, assign to a variable called names, and sort the dataset by highest to lowest count.

In [7]:
names = df.groupby('Name')['Count']\
          .sum()\
          .sort_values(ascending=False)\
          .reset_index()
names

Name   Count
0           Jacob  242874
1            Emma  214852
2         Michael  214405
3           Ethan  209277
4        Isabella  204798
...           ...     ...
17627      Eniola       5
17628    Atlantis       5
17629       Marci       5
17630  Simarpreet       5
17631        Nita       5

[17632 rows x 2 columns]

### 4. How many different names exist in the dataset?

In [8]:
df['Name'].nunique()

17632

**There are 17,632 different names in the dataset.**

### 5. What is the name with most occurrences?

In [9]:
names.head(1)

Name   Count
0  Jacob  242874

**Jacob is the name with the most occurrences.**

### 6. What is the standard deviation of count of names?

In [10]:
names['Count'].std()

11006.0694678915

### 7. Get a summary of the dataset with the mean, min, max, std and quartiles.

In [11]:
df.describe().round(2)

Year       Count
count  1016395.00  1016395.00
mean      2009.05       34.85
std          3.14       97.40
min       2004.00        5.00
25%       2006.00        7.00
50%       2009.00       11.00
75%       2012.00       26.00
max       2014.00     4167.00